### PR TITLE
[aziotctl check] Add supported platform check

### PR DESCRIPTION
**Pull request by daprilik**
Opening as draft while I solicit feedback on details such as warning messages, what platforms constitute which "Tier", and confirming that the implementation actually works as intended across multiple distros.

Two important notes:

- Please ignore the current Tier lists in the code. These are just placeholders until we nail down the tiers.
- *I have not tested the `os_info` library across all distros yet - it may or may not be working as intended!*

### CODE REVIEW

File: aziotctl/src/internal/check/checks/supported_platform.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            | Type::Redhat
+            | Type::RedHatEnterprise
+            | Type::SUSE => Tier::Tier2,
+            _ => Tier::Tier3,
+        };
+
+        Ok(match tier {
+            Tier::Tier1 => CheckResult::Ok,
+            Tier::Tier2 => CheckResult::Warning(anyhow!(
+                "Running on a partially supported, Tier 2 platform ({}). lorem ipsum",
```
Comment (lt72) [inline aziotctl/src/internal/check/checks/supported_platform.rs:80]: Can we add the URL to the tiering docs?

Comment (lt72) [reply aziotctl/src/internal/check/checks/supported_platform.rs:80]: https://docs.microsoft.com/en-us/azure/iot-edge/support?view=iotedge-2018-06

Comment (lt72) [reply aziotctl/src/internal/check/checks/supported_platform.rs:80]: You should check with Kelly if that is the URL to use or if there a lasting one that can be redirected as needed

Comment (arsing) [reply aziotctl/src/internal/check/checks/supported_platform.rs:80]: We need an aziot equivalent of https://aka.ms/iotedge-platsup . I'll make https://aka.ms/aziot-platsup for now and we can redirect it to aziot-specific documentation later (same approach we used for https://github.com/Azure/iot-identity-service/pull/179#discussion_r600971063 )

Comment (arsing) [reply aziotctl/src/internal/check/checks/supported_platform.rs:80]: Done. @daprilik Use https://aka.ms/aziot-platsup

Comment (daprilik) [reply aziotctl/src/internal/check/checks/supported_platform.rs:80]: Awesome. Didn't know we already had this list. I'll update the PR tomorrow.

**Pull request closed without merging**